### DISCUSSION TITLE
Filter out of text and get output as std::string?

### DISCUSSION THREAD

**Discussion by dontknowanything23** [nlohmann/json#2996] [Q&A] [answered]
Hello, 
got an API call wich gives me json data as text back. 
I am trying to get one specific value out of this output. 

My output from the call is sth like this: { "moderated":false,"color":"FFFFFF","key":177592006, .......}
**I want that the std::string output contains -> 177592006**

```
nlohmann::json j = response.text;
std::string output = j["key"];
```

**Comment by nlohmann** [comment]
You need to parse the string first.

**Reply by dontknowanything23** [reply]
Hello, 
thx for the response. 

I've got this now but got this **error**: "JSON_THROW(type_error::create(302, "type must be string, but is " + std::string(j.type_name())));"
_My code now:_

```
	nlohmann::json j = nlohmann::json::parse(response.text);
	std::string output = j["key"];
```

**Reply by nlohmann** [reply]
Then the type is not a string :-)

The library does not provide an automatic conversion. Is it a number?

**Reply by dontknowanything23** [reply]
No its application/json data called with [cpr](https://github.com/whoshuu/cpr) from an API POST. 

Tried to put the output into a buffer string:

```
	std::string buffer = response.text;
	nlohmann::json j = nlohmann::json::parse(buffer);
	std::string output = j["key"];

```
but same error...

**Reply by gregmarr** [reply]
Yes, it is a number, according to your example:

    { "moderated":false,"color":"FFFFFF","key":177592006, .......}

`j["key"]` is `177592006` which is a number, not a string, and so you get a type error when you try to extract it as a string.

**Reply by dontknowanything23** [reply]
and how to get it into a string? tried `static_cast<std::string>(j["key"])`. but didnt worked, same error.

**Reply by gregmarr** [reply]
That's outside of the scope of this library.  You need to do something like `int key = j["key"];` to get the data, and then if you need it as a string, use your favorite number to string conversion method.

**Reply by nlohmann** [reply]
For numbers, the `dump()` function gives you a string representation.

**Comment by dontknowanything23** [comment, marked as answer]
Thanks for your tips and help. Much apreciated!

This is my solution for the problem. It's working.

```
	static_cast<std::string>(response.text);
	nlohmann::json j = nlohmann::json::parse(response.text);
	std::string output = j["key"];
```
